### PR TITLE
gcylc: fixed definition order sort bug

### LINKED ISSUE #1933: gcylc dot-view "definition order" sort broken on master
On current master "View -> Options -> Toggle definition order" has no effect.

It looks like this was broken by #1881 - @oliver-sanders can you take a look please?

From a quick look, you moved the sorting code up a bit, and as a result the sorted `self.task_list` immediately gets overwritten:

https://github.com/cylc/cylc/pull/1881/files#diff-206ad852431585eaedb8db12e16ff1daR130

However, simply moving the sorting block to the end of the method doesn't work: playing with `examples/task-states` family collapse in the transpose view eats too many tasks (more than just the family members).

### PR DESCRIPTION
**Pull request by oliver-sanders**
Closes #1933 

@hjoliver Please review.
@benfitzpatrick Please review.

### CODE REVIEW

**Comment by hjoliver** [comment]
@oliver-sanders - this isn't working when families are present.  E.g. for this suite:
```
[scheduling]
    [[dependencies]]
        graph = a & b & c & d & e & f & FAM
[runtime]
    [[FAM]]
    [[z]]
        inherit = FAM
    [[y]]
        inherit = FAM
    [[x]]
        inherit = FAM
    [[f]]
    [[e]]
    [[d]]
    [[c]]
    [[b]]
    [[a]]
```
The correct sort order is shown in the right, in the tree view:
foo

**Comment by oliver-sanders** [comment]
@hjoliver Sorry about that, should be working now.

**Comment by hjoliver** [comment]
Thanks, that's got it.  Review 1 - good, over to @benfitzpatrick

File: lib/cylc/gui/updater_dot.py [hunk 1]
```diff
@@ -127,45 +127,44 @@ def update(self):
...
-                                  tasks + self.task_list[ind - 1:])
-
-            # Remove duplicates but maintain order.
-            tasks = []
-            for i, task in enumerate(self.task_list):
-                if task not in tasks:
-                    tasks.append(task)
-            self.task_list = tasks
-
+                self.task_list = (self.task_list[0:ind] + [heading] + tasks +
```
Comment (benfitzpatrick) [inline lib/cylc/gui/updater_dot.py]: That's interesting - was it doing the wrong thing before?

Comment (oliver-sanders) [inline lib/cylc/gui/updater_dot.py]: It was ordering the list the other way around.

**Pull request merged**